### PR TITLE
marsRover19: Adds passport config, cookie extraction and authentication to /userAccountPage

### PR DESCRIPTION
**Pull request by HeatherJLewis**
**Description**
Includes the work for the following:
* marsRover19: create passport config
* marsRover20: create an account endpoint and wrap in passport authentication

**Changes**
* Adds passport config
* Adds `cookieExtractor` function to get the JWT from the `access_token` cookie
* Adds passport authentication to `/userAccountPage`

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const passport = require('passport');
+const JwtStrategy = require('passport-jwt').Strategy;
+const { PRIVATE_KEY } = require('../config/apiCredentials');
+
+const cookieExtractor = (request) => {
+	var token = null;
+
+	if (request && request.headers.cookie) {
+		token = request.headers.cookie.split('=')[1];
```
Comment (Kenneth-Siu) [inline config/passport.js]: What would happen here in the future if we use cookies for other reasons? Would this always work?

(And incidentally, there's an instance of a magic number here, but fixing my main question will hopefully end up removing the magic number anyway)

File: config/passport.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+	return token;
+};
+
+let options = {};
+options.jwtFromRequest = cookieExtractor;
+options.secretOrKey = PRIVATE_KEY;
```
Comment (Kenneth-Siu) [inline config/passport.js]: When it comes to defining an object and them immediately defining its properties, I like to put them together in one declaration.

File: config/passport.js (hunk 1 above)
Comment (Kenneth-Siu) [reply config/passport.js]: (I believe there's a library mentioned on the passport docs that's quite useful, to do with cookies)

**Review by Kenneth-Siu — COMMENTED** [review]
Yay I'm so happy when all the authentication pieces come together. It's always a bit of a steep climb!

A couple of thoughts here, but everything looks neat and tidy as usual :)

File: config/passport.js (hunk 2 above)
Comment (HeatherJLewis) [reply config/passport.js]: Yes, I thought I'd fixed this. Will sort this now. :)

Comment (HeatherJLewis) [reply config/passport.js]: Done

File: config/passport.js (hunk 1 above)
Comment (HeatherJLewis) [reply config/passport.js]: Added cookie-parser and updated code to use this.

**Review by Kenneth-Siu — APPROVED** [review]
Nice looks ace 👌

**Pull request merged**